### PR TITLE
Flip default on weights_only

### PR DESCRIPTION
**Pull request by mikaylagawarecki**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #139303
* __->__ #137602
* #139541
* #139433
* #139221
* #138936

cc @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @rec

### CODE REVIEW

File: .github/ci_commit_pins/xla.txt [hunk 1]
```diff
@@ -1 +1 @@
-2eb4a60ed14a38260b85b0c765161f0ce45be6d1
+f71c02d1f457d58371e013632efb016c01bd1866
```
Comment (mikaylagawarecki) [inline .github/ci_commit_pins/xla.txt:1]: This was intentional to include  https://github.com/pytorch/xla/pull/8259

**Comment by kit1980** [comment]
I'm not sure if it's OK to use different default in fbcode/OSS.
cc @malfet

**Comment by kit1980** [comment]
Also this will break a lot of people.
Maybe it's time to break.

File: .github/ci_commit_pins/xla.txt (hunk 1 above)
Comment (malfet) [inline .github/ci_commit_pins/xla.txt:1]: This is not intended, is it?

File: torch/serialization.py [hunk 2]
```diff
@@ -1159,7 +1160,7 @@ def load(
     map_location: MAP_LOCATION = None,
     pickle_module: Any = None,
     *,
-    weights_only: Optional[bool] = None,
+    weights_only: Optional[bool] = not IS_FBCODE,
```
Comment (malfet) [inline torch/serialization.py:1163]: if you are changing it to hard boolean, please remove `Optional` and delete the logic associated with opt-in checks
```suggestion
    weights_only: bool = not IS_FBCODE,
```

**Review by malfet — APPROVED** [review]
Tentatively LGTM, but please change the signatures a bit further. And `not IS_FBCODE` feels a bit suspicious, perhaps better behavior would be to add `weights_only=False` to respective callsights?

File: .github/ci_commit_pins/xla.txt (hunk 1 above)
Comment (mikaylagawarecki) [reply .github/ci_commit_pins/xla.txt:1]: This is intended, see https://github.com/pytorch/pytorch/pull/137602#discussion_r1803427860

**Review by albanD — APPROVED** [review]
A couple more pieces need to be added to the CI and we should announce this one as a bc-breaking change broadly for sure!
But yes let's do it!

**Pull request closed without merging**